benchmarks/run_pytest_matrix.py

```
@@ -0,0 +1,223 @@
+        [str(python), "-c", "import pytest; print(pytest.__version__)"],
+        cwd=REPO_ROOT,
+        text=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        check=True,
+    )
+    return result.stdout.strip()
+
+
+def fetch_latest_pytest_version() -> str:
+    with urllib.request.urlopen(PYPI_JSON_URL, timeout=15) as response:
```

Ruff (0.15.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)